feat: fetch url meta data on creating row

zoli opened the pull request:
On submitting a new row in board view, if the setting is enabled and the text is a URL fetch url metadata and fill the row with it.

closes #5963

### Feature Preview


https://github.com/user-attachments/assets/26ffc1c1-e009-4079-be28-a1c92c3a1861



#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

zoli commented:
At first, I wanted to also enable this feature in grid view on filling URL field, but filling other fields inside `EditableURLCell` isn't that straightforward. So I put it aside for maybe in the future.

The next step can be adding the description and image of URL metadata in the row page.

LucasXu0 commented:
After inputting the URL, there's a noticeable delay before the parsed URL is inserted (I understand this is due to fetching metadata). This lag affects the user experience. We should consider a better way to handle this. For example:

1. insert the raw URL text immediately.
2. create a new item with the raw URL text.
3. update the cell content with parsed metadata once it's available.

@zoli What do you think?

Screenshots: 
<img width="201" alt="Screenshot 2024-09-09 at 09 34 59" src="https://github.com/user-attachments/assets/ad2d3e9a-88f2-4b93-aab0-0b6042d124e6">
<img width="215" alt="Screenshot 2024-09-09 at 09 35 04" src="https://github.com/user-attachments/assets/f8891648-b87d-4571-99c9-da526862d968">
<img width="201" alt="Screenshot 2024-09-09 at 09 35 10" src="https://github.com/user-attachments/assets/26c4fd6d-c909-4f7a-b257-c6ec3e164632">

zoli commented:
@LucasXu0 By creating a new item do you mean sending `CreateRow` event to the backend? I think creating a pseudo card while fetching the URL is better, Maybe with less opacity so the user knows it's not final yet.

richardshiue commented:
What should the behavior be when there are more than one URL fields?

zoli commented:
> What should the behavior be when there are more than one URL fields?

I can change the settings to choose a specific field, for now, I thought selecting the first URL field type would be good enough.

zoli commented:
@richardshiue will choosing specific field in settings and pseudo card while fetching the URL be enough for this?

appflowy commented:
Hi @zoli , please add bloc tests or integration test.

zoli commented:
@appflowy, Added both bloc and integration tests.
@richardshiue, Enabled choosing the field (URL type) to fill the URL.
@LucasXu0,  Added loading state.

LucasXu0 commented:
@richardshiue What do you think about it? I'm thinking that the toggle function should be an option for the URL field type, not in the global settings of the database.
